### PR TITLE
Add twist_mux node

### LINKED ISSUE #3: Create twist_mux node
Inputs: Twist message, joy
Outputs: Twist message

The twist_mux node will act as a selector of twist messages.  It will be in charge of controlling where the twist message which is being sent to the motor is coming from (i.e. Autonomous or teleop).

### PR DESCRIPTION
**Pull request by matthewia94**
The twist_mux node is a node which takes in multiple twist messages and
publishes only a single twist message to the motor controller on the
"cmd_vel" topic.  The node currently uses the buttons defined in the
constants.h file.

Closes #3

### CODE REVIEW

File: ros_ws/src/common/constants.h [hunk 1]
```diff
@@ -22,15 +38,27 @@ const int X_AXIS_R_STICK = 2;
...
  */
 const std::string TELEOP_TOPIC = "teleop_twist";
+/**
+ * @brief The topic that autonomous twist messages will be published on
+ */
+const std::string AUTO_TOPIC = "auto_twist";
+/**
+ * @brief The topic containing twist messages to be sent to the motor controller
+ */
+const std::string CMD_VEL_TOPIC = "cmd_vel";
```
Comment (islamelnabarawy) [inline ros_ws/src/common/constants.h]: Let's rename CMD_VEL_TOPIC to be something more generic so we wouldn't be locked into using the particular topic called "cmd_vel". Maybe CONTROL_TOPIC or something similar...

File: ros_ws/src/control/src/control/twist_mux.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+void TwistMux::twistCallback(const geometry_msgs::Twist::ConstPtr& msg)
+{
+    cmd_vel.linear.x = msg->linear.x;
+    cmd_vel.linear.y = msg->linear.y;
+    cmd_vel.linear.z = msg->linear.z;
+    cmd_vel.angular.x = msg->angular.x;
+    cmd_vel.angular.y = msg->angular.y;
+    cmd_vel.angular.z = msg->angular.z;
```
Comment (islamelnabarawy) [inline ros_ws/src/control/src/control/twist_mux.cpp]: You don't have to manually copy every single member. Just doing the following assignment should be enough:

cmd_vel = *msg;

File: ros_ws/src/control/src/control/twist_mux.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+void TwistMux::stopRobot()
+{
+    cmd_vel.linear.x = 0;
+    cmd_vel.linear.y = 0;
+    cmd_vel.linear.z = 0;
+    cmd_vel.angular.x = 0;
+    cmd_vel.angular.y = 0;
+    cmd_vel.angular.z = 0;
```
Comment (islamelnabarawy) [inline ros_ws/src/control/src/control/twist_mux.cpp]: Similarly, I believe assigning an empty Twist will accomplish the same thing here:

cmd_vel = geometry_msgs::Twist();

File: ros_ws/src/control/include/control/twist_mux.h [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+private:
+    /**
+     * @brief An enumerator to describe the mode the robot is currently in
+     */
+    enum Mode
+    {
+        standby,        //!< The robot is not moving, the same as a software e-stop
+        teleop,         //!< The robot is currently being controlled by remote control
+        autonomous      //!< The robot is driving itself
+    };
```
Comment (islamelnabarawy) [inline ros_ws/src/control/include/control/twist_mux.h:33]: This (unscoped) enum will pollute the global namespace with the names of its members.

Please see [the ROS C++ Style Guide section on Enumerations](http://wiki.ros.org/CppStyleGuide#Enumerations) for the proper way to define enums without polluting the global namespace.

**Comment by matthewia94** [comment]
This also addresses issue #4

**Pull request merged**